File: BinarySearchTree/151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {

        string token = "";
        stringstream ss(s);
        vector<string> word;
        while (ss >> token) {
            word.push_back(token);
        }

        string ans = "";
        for (int i = word.size() - 1; i >= 0; i--) {
            ans += word[i];
            if (i != 0)
                ans += ' ';
        }
        return ans;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        nums=list(s)
        left=0
        right=len(nums)-1
        while left<=right:
                nums[left],nums[right]=nums[right],nums[left]
                left+=1
                right-=1 
        nums.append(' ')

        ans=""
        x=0
        while x<len(nums):
           while x<len(nums) and nums[x]==' ':
                x+=1

           temp=""
           while x<len(nums) and nums[x]!=' ' :   
                temp+=nums[x]
                x+=1

           temp=temp[::-1]
           if temp:
              ans+=temp+' '     

        return ans.strip()           